src/components/download/downloadButtons.js

```diff
@@ -35,6 +36,10 @@ export const DownloadButtons = ({dispatch, t, tree, entropy, metadata, colorBy,
       }
     }
   }
+  /* Verify that we can parse dataset name from URL to support Auspice JSON download */
+  const datasetNames = getDatasetNamesFromUrl(window.location.pathname);
+  const supportAuspiceJsonDownload = !gisaidProvenance && datasetNames.some(Boolean);
```

Comment:
There's a bug in auspice.us (?) where dragging on two trees results in them being added to the URL. As such, `datasetNames` contains 2 truthy values, and we allow the download, which then fails.

Using `dev_multi_h3n2.json` and `ebola.json` (any two JSONs will do) we get a URL of `https://auspice-us-nextstrain-b-q2kkxc.herokuapp.com/dev/multi/h3n2:ebola`, and the following error when trying to download:

<img width="492" alt="image" src="https://github.com/user-attachments/assets/4727d9ab-a2d4-4b4f-b630-9e33adb2886a">

Reply:
Oh, I did not realize that the URL gets updated in auspice.us when there's a two trees. 
Looks like it's coming from the `changeURLMiddleware` in auspice:

https://github.com/nextstrain/auspice/blob/69557289fe7b4a51c4d529c97ee02fea1cafa6a9/src/middleware/changeURL.js#L236-L243

I assume the URL should never be updated in auspice.us if you are calling this a bug? How is that configured?

Reply:
Hmm, maybe we just shouldn't rely on the URL parsing here to determine if download is supported. We can add a `disableAuspiceJsonDownload` config param to the extensions and have auspice.us explicitly disable the downloads.

Reply:
> I assume the URL should never be updated in auspice.us if you are calling this a bug? How is that configured?

auspice.us, once it's parsed the dropped files, [initialises the Auspice viz via:](https://github.com/nextstrain/auspice.us/blob/96c8d5486f2852f2fa66aa995ee30a637efc5230/auspice_client_customisation/handleDroppedFiles.js#L200-L205)

```js
  /* load (into redux state) the main dataset(s) */
  try {
    dispatch({
      type: "CLEAN_START",
      pathnameShouldBe: "",
      ...createStateFromQueryOrJSONs({
```

and the intention is that Auspice updates the pathname (URL) to the empty string via:

https://github.com/nextstrain/auspice/blob/69557289fe7b4a51c4d529c97ee02fea1cafa6a9/src/middleware/changeURL.js#L231-L234

and then the switch statement ends. But `""` is falsey, so we keep going in the switch statement and then, when there are two trees, we hit the case you linked above. I think checking if `pathnameShouldBe` is a property on the action rather than just the truthyness of its value is the correct fix.

> Hmm, maybe we just shouldn't rely on the URL parsing here to determine if download is supported. We can add a `disableAuspiceJsonDownload` config param to the extensions and have auspice.us explicitly disable the downloads.

I think once the bugs identified here are fixed the URL parsing will still be appropriate. (The particular bug in this thread is, strictly speaking, separate to this PR, but it'd be good to just fix it here.)

Reply:
Updated in https://github.com/nextstrain/auspice/pull/1804/commits/a9c461057f9258b88aa9e8882f1664cb2238b7e6.

Reply:
Confirmed in [auspice.us test app](https://auspice-us-nextstrain-b-q2kkxc.herokuapp.com/) that the URL does not update with multiple trees.